### ISSUE TITLE
Incompatibilities between v1.21 and v1.24 ?

### ISSUE THREAD

**Issue by kbaniszack** [veracrypt/VeraCrypt#535]
Hi and thank you for this great tool.

I used to use version 1.21 of veracrypt on Windows.
I updated to version 1.24 hotfix 1 but I could not open my encrypted containers ("bad password or no encrypted container").
I uninstalled version 1.24 hotfix 1 to release version 1.21 and the containers are working again.
I tried version 1.24, same problem.

I create a container via version 1.24 hotfix 1 under linux and can not open it under windows version 1.21 ("bad password or not an encrypted container")

**Comment by idrassi** [comment]
Are you using a password longer than 64 characters?

Before version 1.24, maximum password length was 64 characters and so VeraCrypt truncated longer password to their first 64 characters (in version 1.23, we displayed a warning when truncation happened).
Starting from version 1.24, maximum password length was increased to 128 characters and so truncation don’t happen, thus if the user enters the long password in VeraCrypt 1.24, mounting will fails.

On Windows an options was added on VeraCrypt 1.24 Preferences to force the use of the old maximum length (64) so that the truncation happens again and mount succeeds. Below is a screenshot of this option.

![VeraCrypt_Preferences_PassLen](https://user-images.githubusercontent.com/4698669/68031678-6af4f380-fcbc-11e9-89be-1070c61952eb.png)
